bundles/org.openhab.binding.ism8/src/main/java/org/openhab/binding/ism8/server/Server.java

```diff
@@ -0,0 +1,281 @@
+                            for (int k = 0; k < this.dataPoints.size(); k++) {
+                                IDataPoint dp = this.dataPoints.get(k); 
+                                if (dp.getId() == messages[j].getId()) {
+                                    dataPoint = dp;
+                                    break;
+                                }
+                            }
+
+                            if (dataPoint != null) {
+                                dataPoint.processData(messages[j].getData());
+                                logger.debug("{} {}", dataPoint.getDescription(), dataPoint.getValueText());
+                                if (listener != null) {
+                                    listener.dataPointChanged(new DataPointChangedEvent(this, dataPoint));
+                                }
+                                break;
+                            }
+                        }
+                    }
+                }
+            }
+        } catch (Exception e) {
+            logger.warn("Error handle client data stream. {}", e.getMessage(), e);
+            this.stopServer();
+        }
+    }
+
+    private void sendUpdateCommand() throws Exception {
+        byte[] data = new byte[] { (byte) 0x06, (byte) 0x20, (byte) 0xF0, (byte) 0x80, (byte) 0x00, (byte) 0x16,
+                (byte) 0x04, (byte) 0x00, (byte) 0x00, (byte) 0x00, (byte) 0xF0, (byte) 0xD0 };
+        this.sendData(data);
+    }
+
+    private byte[] getBytesFromInputStream(InputStream is) throws IOException {
+        ByteArrayOutputStream os = new ByteArrayOutputStream();
+        byte[] buffer = new byte[0xFFFF];
+        int len = is.read(buffer);
+        os.write(buffer, 0, len);
+        return os.toByteArray();
+    }
+
+    private ArrayList<byte[]> getPackages(byte[] data, int amount) {
+        ArrayList<byte[]> result = new ArrayList<byte[]>();
+        if (data.length >= amount) {
+            ByteBuffer list = ByteBuffer.allocate(amount);
+            for (int i = 0; i < amount; i++) {
+                list.put(data[i]);
+            }
+
+            int start = -1;
+            for (int i = 0; i < amount - 4; i++) {
+                if (list.get(i + 0) == (byte) 0x06 && list.get(i + 1) == (byte) 0x20 && list.get(i + 2) == (byte) 0xF0
+                        && list.get(i + 3) == (byte) 0x80) {
+                    if (start >= 0) {
+                        byte[] pkgData = new byte[i - start];
+                        for (int j = 0; j < pkgData.length; j++) {
+                            pkgData[j] = list.get(start + j);
+                        }
+                        result.add(pkgData);
+                    }
+                    start = i;
+                }
+            }
+            if (start >= 0) {
+                byte[] pkgData = new byte[amount - start];
+                for (int j = 0; j < pkgData.length; j++) {
+                    pkgData[j] = list.get(start + j);
+                }
```

Comment:
You could take a look at the more convenient method `list.get(dst, offset, length)`. Same for above.

Reply:
the method `list.get(dst, offset, length)` does not work as the current code. In case of replacing the existing code above into:
`byte[] pkgData = new byte[amount - start];`
`list.get(pkgData, start, pkgData.length);`
then I'm getting an BufferUnderflowException

Reply:
The `offset`/`start` is the offset in the destination array. You need to position the ByteBuffer with `list.position(start)` before and then use `list.get(pkgData, 0, pkgData.length)`.

Reply:
OK, thanks - this works. I thought this is the offset in the list, but not in the destination.

Reply:
You could simply use `list.get(pkgData)` with `list.position(start)` beforehand.

Reply:
I had changed those simplifications as well as the put. Connected to this change I could simplyfy this method and reduce the method parameter.